.github/workflows/ci.yml

```diff
@@ -252,25 +249,17 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - name: install node for Pyright
-      uses: actions/setup-node@v4
-      with:
-        node-version: '20'
-
```

Comment:
Let's align with the changes from the previous PR, and use the pyright PyPI wrapper as well. As a consequence, the `typechecking` deps grp has pyright added.